module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -567,7 +607,235 @@ protected function getSupportedCitationFormats()
      */
     public function exportDisabled($format)
     {
-        // EDS is not export-friendly; disable all formats.
-        return true;
+        // at the moment we only support RIS
+        return strtolower($format) != 'ris';
+    }
+    
+     /**
+     * This method can fetch fields out of the deeply nested RecordInfo array. 
+     * many fields are here in a more readable and atomic format
+     * 
+     * The latter two params are only for internal use (recursion)
+     *  
+     * @param array $arrayKeys Key path to the needed value
+     * @param int   $level     only used for recursion
+     * @param array $fields    only used for recursion
+      * 
+     * @return array|string
+     */
+    protected function getFieldRecursive($arrayKeys, $level = 0, $fields = null)
+    {
+        // special case: we have to iterate through items array
+        if ($arrayKeys[0] == 'Items' && isset($arrayKeys[1])) {
+            $items = $this->getItems($arrayKeys[1]);
+            $output = [];
+            foreach ($items as $item) {
+                $output[] = $item['Data'];
+            }
+            return implode('; ', $output);
+
+        }
+
+        if (!$fields) {
+            $fields = $this->fields;
+        }
+        
+        if (isset($fields[$arrayKeys[$level]])) {
+            $newFields = $fields[$arrayKeys[$level]];
+            $level++;
+            if ($level < count($arrayKeys)) {
+                return $this->getFieldRecursive($arrayKeys, $level, $newFields);
+            } else {
+                // end of recursion
+                return $newFields;
+            }
+        }
+        return '';
+    }
+    
+    /**
+     * Tries to get a piece of information from different locations, configured
+     * in the prioritizedFields property. 
+     * 
+     * @param string $method array key from $this-prioritizedFields
+     * 
+     * @return string
+     */
+    protected function prioritizedFields($method)
+    {
+        $result = '';
+        if (isset($this->prioritizedFields[$method])) {
+            foreach ($this->prioritizedFields[$method] as $prio => $arrayKeys) {
+                if (empty($result)) {
+                    $result = $this->getFieldRecursive($arrayKeys);
+                }
+            }
+
+        }
+        return $result;
+    }
+    
+    /**
+     * Get title of containing record
+     * 
+     * @return string
+     */
+    public function getContainerTitle()
+    {
+        return $this->prioritizedFields('ContainerTitle');
+    }
+    /**
+     * Get issue of containing record
+     * 
+     * @return string
+     */
+    public function getContainerIssue()
+    {
+        $numbering = $this->prioritizedFields('ContainerIssue');
+        if (is_array($numbering)) {
```

Comment:
Do you think a getPrioritizedFieldDataOfType method might eliminate some of this redundant logic throughout the class? It could take a field name and an array of types, and return all the non-empty values that match the type array. Then for methods that require just one value, we could take the first item from the array, and for multi-valued methods, we could return the array as-is. I think that might make this whole class a lot shorter and more readable.